lib/screens/v2/candidate_generator/widgets/reconstructed_alert.ex

```diff
@@ -23,32 +22,24 @@ defmodule Screens.V2.CandidateGenerator.Widgets.ReconstructedAlert do
           app_params: %PreFare{reconstructed_alert_widget: %CurrentStopId{stop_id: stop_id}}
         } = config,
         now \\ DateTime.utc_now(),
-        fetch_routes_by_stop_fn \\ &Route.fetch_routes_by_stop/3,
-        fetch_stop_sequences_by_stop_fn \\ &RoutePattern.fetch_parent_station_sequences_through_stop/2,
         fetch_alerts_fn \\ &Alert.fetch/1,
-        fetch_stop_name_fn \\ &Stop.fetch_stop_name/1
+        fetch_stop_name_fn \\ &Stop.fetch_stop_name/1,
+        fetch_location_context_fn \\ &Stop.fetch_location_context/3
       ) do
-    # Filtering by subway and light_rail types
-    with {:ok, routes_at_stop} <- fetch_routes_by_stop_fn.(stop_id, now, [:light_rail, :subway]),
-         route_ids_at_stop =
-           routes_at_stop
-           |> Enum.map(& &1.route_id)
-           # We shouldn't handle Mattapan outages at this time
-           |> Enum.reject(fn id -> id === "Mattapan" end),
-         {:ok, alerts} <- fetch_alerts_fn.(route_ids: route_ids_at_stop),
-         {:ok, stop_sequences} <-
-           fetch_stop_sequences_by_stop_fn.(stop_id, route_ids_at_stop) do
+    with {:ok, location_context} <- fetch_location_context_fn.(PreFare, stop_id, now),
+         # TODO: make sure we're handling Mattapan outages correctly even without the "reject"
+         {:ok, alerts} <- fetch_alerts_fn.(route_ids: location_context.route_ids_at_stop) do
       alerts
-      |> Enum.filter(&relevant?(&1, config, stop_sequences, routes_at_stop, now))
+      |> Enum.filter(&relevant?(&1, config, location_context, now))
       |> Enum.map(fn alert ->
         %ReconstructedAlert{
           screen: config,
           alert: alert,
           now: now,
-          stop_sequences: stop_sequences,
-          routes_at_stop: routes_at_stop,
+          location_context: location_context,
+          # TODO: these two items look like location stuff? Should they be?
```

Comment:
For `informed_stations_string`, I think no. It is getting info based on the `informed_entities` so it makes sense to keep that separate. `is_terminal_station` might be a good thing to have in the `location_context` though. The logic for the function should be the same everywhere so moving that to `fetch_location_context` seems safe.

Reply:
On second thought, `is_terminal_station` is only used by Pre-fare alerts right now, so maybe I won't add it to LocationContext unless it becomes necessary for more screens later?

Reply:
That's fair. I don't see that level of context being important for any other alert widget.